From 98ac5e8b165a461dad06edd3940d4963cd5292af Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Thu, 23 Nov 2023 11:05:20 -0500
Subject: [PATCH 23/24] HACK: Ignore errors when setting fscrypt directory.
 Some devices fail to set it on first boot, but is happy later (?!?)

---
 init/builtins.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/init/builtins.cpp b/init/builtins.cpp
index bc23972a2..f2944ebca 100644
--- a/init/builtins.cpp
+++ b/init/builtins.cpp
@@ -428,9 +428,13 @@ static Result<void> make_dir_with_options(const MkdirOptions& options) {
         }
     }
     if (IsFbeEnabled()) {
+        //TODO FIXME
+//01-01 06:10:54.928   477   477 E vold    : Failed to set encryption policy of /data/data to be0c03668ef1fddc v1 modes 127/4 flags 0x0: Invalid argument
+//01-01 06:10:55.927     0     0 W         : c2    477 ext4_process_policy: Policy inconsistent with encryption context
+//01-01 06:10:55.927     0     0 E         : c2    477 vold: Failed to set encryption policy of /data/data to be0c03668ef1fddc v1 modes 127/4 flags 0x0: Invalid argument
+
         if (!FscryptSetDirectoryPolicy(ref_basename, options.fscrypt_action, options.target)) {
-            return reboot_into_recovery(
-                    {"--prompt_and_wipe_data", "--reason=set_policy_failed:"s + options.target});
+            return {};
         }
     }
     return {};
-- 
2.25.1

